File: Compilers/Lab8/src/lexer/Fragment.java
package lexer;

public class Fragment {
    private Position start;
    private Position end;

    public Fragment(Position start, Position end) {
        this.start = start;
        this.end = end;
    }

    @Override
    public String toString() {
        return start.toString() + "-" + end.toString();
    }
}

File: Optimal Code Generation/Lab3/Lexer/Fragment.cpp
//
// Created by alexis on 06.06.2020.
//

#include "Fragment.h"
#include <iostream>

Fragment:: Fragment(const Position &start, const Position &end):
        start(start), end(end) {}


void Fragment::print() {
    cout << "(" << start.getLine() << " ," << start.getPosition() << ") - (" << end.getLine() << " ," << end.getPosition() << ")";
}